sklearn/tree/_tree.pyx

```diff
@@ -1292,8 +1292,8 @@ cdef _cost_complexity_prune(unsigned char[:] leaves_in_subtree, # OUT
     """Perform cost complexity pruning.
 
     This function takes an already grown tree, `orig_tree` and outputs a
-    boolean mask `leaves_in_subtree` to are the leaves in the pruned tree. The
-    controller signals when the pruning should stop and is passed the
+    boolean mask `leaves_in_subtree` which are the leaves in the pruned tree. The
+    controller signals when the pruning should stop and is passed wtih the
```

Comment:
In this case, the controller is passed the metrics of the subtrees. Thus I think the original phrasing makes sense.

```suggestion
    controller signals when the pruning should stop and is passed the
```

Reply:
I still don't understand this sentence indeed. From what I can see, the controller is used to signal when the pruning stop and to store the metric before and after pruning?

Reply:
> From what I can see, the controller is used to signal when the pruning stop and to store the metric before and after pruning?

Yea the controller is doing quite a bit. It could be worth explaining **why** the controller needs this information.